### PR TITLE
Bump minimum required PHP version to 7.2.

### LINKED ISSUE #60680: Increase the PHP minimum supported version to 7.2
On April 8th, WordPress announced it is dropping PHP 7.0 and 7.1 support, raising the minimum supported version to PHP 7.2 for WordPress 6.6.0.

https://make.wordpress.org/core/2024/04/08/dropping-support-for-php-7-1/

## What is your proposed solution?

To keep parity with WordPress Core, this ticket proposes for Gutenberg to:

* Raise the minimum support PHP version.
* Drop the PHP 7.0 and 7.1 CI jobs.

## Tasks

1. [x] Drop PHP 7.0 and 7.1 CI jobs running on WordPress trunk.
    PR: https://github.com/WordPress/gutenberg/pull/60686#top
2. [x] Get consensus on when to raise the minimum required PHP version.
3. [x] Change the plugin header `Requires PHP:` to `7.2`.

### PR DESCRIPTION
**Pull request by anton-vlasenko**
## What?

* Bumps the plugin's minimum PHP version from 7.0 to 7.2.
* Bumps PHPCS `testVersion` from `7.0-` to `7.2-`.
* Bumps the PHPUnit test `Test with previous WP version` from `php: 7.0` to `php: 7.2`.

Fixes https://github.com/WordPress/gutenberg/issues/60680.

## Why?

WordPress Core parity.

[WordPress 6.6.0's minimum supported PHP version is/will be 7.2](https://make.wordpress.org/core/2024/04/08/dropping-support-for-php-7-1/).

## How?

Modifies the plugin header's `Requires PHP:` from `7.0` to `7.2`.

## Testing Instructions
1. Change your local test environment to PHP 7.0 (7.1) and WordPress 6.6-alpha.
2. Install and activate the Gutenberg 18.1.x plugin.
3. Open the `gutenberg.php` file, and in the file's DocBlock change `Requires PHP:` back to `7.2`.
Expected: Stays activated, bulk actions disabled, and a notice appears:
>This plugin does not work with your version of PHP. [Learn more about updating PHP](https://wordpress.org/support/update-php/).
4. Deactivate the plugin.
Expected: Bulk edit is disabled; `Cannot Activate` is displayed but not clickable; notice appears:
>This plugin does not work with your version of PHP. [Learn more about updating PHP](https://wordpress.org/support/update-php/).

### CODE REVIEW

**Comment by anton-vlasenko** [comment]
~~Let's not merge this one now; I need some time to test it to ensure it works as expected.~~

This question needs to be discussed first: https://github.com/WordPress/gutenberg/issues/60680#issuecomment-2052626690

**Review by johnbillion — APPROVED** [review]

**Review by Soean — APPROVED** [review]

**Comment by anton-vlasenko** [comment]
The PR has received two approvals.
I think this is sufficient consensus, considering the process is similar to https://github.com/WordPress/gutenberg/issues/52344.
I will proceed with the version bump as soon as the CI jobs pass.

**Comment by anton-vlasenko** [comment]
Thank you for approving the PR, @johnbillion  and @Soean!
The version has been bumped.

**Comment by colorful-tones** [comment]
Note: this is being included in the Gutenberg 18.2 release, because there was a conflict during release and it just seemed easier path forward.

**Pull request merged**
